src/bridge/cockpitpackages.c

```diff
@@ -374,12 +374,30 @@ expand_libexec (const gchar *variable,
   return NULL;
 }
 
+static void
+apply_override (JsonObject *manifest,
+                const char *directory,
+                const char *filename)
+{
+  g_autoptr(GError) error = NULL;
+
+  g_autoptr(JsonObject) override = read_json_file (directory, filename, &error);
+  if (error)
+    {
+      if (g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+        g_debug ("no override found in %s/%s", directory, filename);
+      else
+        g_message ("couldn't read %s/%s: %s", directory, filename, error->message);
+    }
+  if (override)
+    cockpit_json_patch (manifest, override);
```

Comment:
*shudders*

The one sole case of mutable JSON in the entire cockpit C codebase, and you're making it worse!

Reply:
How does this make it worse? It's the exact same call, just moved into a helper function?

Reply:
I.e. how would you like to change it? This looks perfectly reasonable to me,  `JsonObject*` is mutable.